src/pipe/pipe.c

```diff
@@ -0,0 +1,132 @@
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include "pipe.h"
+#include "../shared/file_util.h"
+
+#define BUFFSZ 256
+
+int use_pipe(setting_t settings){
+    printf("usp %s\n", settings.password);
+    switch (settings.role) {
+        case SENDER:
+            return pipe_sender(settings);
+            break;
+        case RECEIVER:
+            return pipe_receiver(settings);
+            break;
+        default:
+            err_and_leave("Impossible role", 4);
+    }
+    return 0;
+}
+
+char* fifo_name(char* passwd){
+    printf("ffn %s\n", passwd);
+    char* myfifo = "/tmp/myfifo_";
+    // We have passwd
+    // We have myfifo
+    // new_string
+    // copy myfifo into new_string
+    // concatenate passwd to new_string
+    char* new_string = malloc(strlen(myfifo) + strlen(passwd) + 1);
+    strcpy(new_string, myfifo);
+    strcat(new_string, passwd);
+    return new_string;
+}
+
+void create_fifo(char* passwd){
+    printf("ffn %s\n", passwd);
+    char* new_string = fifo_name(passwd);
+    printf("%s\n", new_string);
+    int ret = mkfifo(new_string, 0666);
+    if (ret == 0){
+    } else if (ret == -1) {
+        switch (errno)
+        {
+        case EEXIST:
+            break;
+        default:
+        
+        printf("%d\n", errno);
+            err_and_leave("Failed to create pipe", 4);
+            break;
+        }
+    } else {
+        err_and_leave("", 4);
+    }
+}
+
+int pipe_sender(setting_t settings){
+    int fd_in, fd_fifo;
+    create_fifo(settings.password);
+    char* fifon = fifo_name(settings.password);
+    fd_fifo = open_file(fifon, O_WRONLY, 0);
+    //printf("Opened\n");
+    fd_in = open_file(settings.filename, O_RDONLY, 0);
+
+    char buffer[BUFFSZ]; // 
+    size_t read_position = 0;
+    FILE* fifo_stream = fdopen(fd_fifo, "w");
+
+    struct stat sb;
+    if (stat(settings.filename, &sb) == -1) {
+        err_and_leave("stat", 4);
+    }
+    size_t size = sb.st_size;
+    while (1) {
+        int n_read = read_chunk(fd_in, read_position, BUFFSZ, buffer, 1);
+        int n_write = fwrite(buffer, 1, n_read, fifo_stream);
+        if (!(n_write==n_read)){
+            err_and_leave("Wrote different amount than read", 4);
+        }
+
+        read_position = read_position + n_read;
+        //printf("Read: %.*s\n", n_read, buffer);
+        if (read_position == size){
+            break;
+        }
+    }
+    fclose(fifo_stream);
+    close(fd_in);
+    return 0;
+}
+
+int pipe_receiver(setting_t settings){
+    int fd_out, fd_fifo;
+    create_fifo(settings.password);
+    char* fifon = fifo_name(settings.password);
+    fd_fifo = open_file(fifon, O_RDONLY, 0);
+    //printf("Opened\n");
+    fd_out = open_file(settings.filename, O_WRONLY | O_APPEND | O_CREAT, S_IWRITE | S_IREAD);
```

Comment:
```suggestion
    fd_out = open_file(settings.filename, O_WRONLY | O_APPEND | O_CREAT | O_TRUNC, S_IWRITE | S_IREAD);
```
This will reset the file if it already exists, alternatively add `O_EXCL` instead of `O_TRUNC` to the flags to fail if the output file exists already

Thanks @millavalio-Unikie, @KocogluFerdi please commit this change